src/actions/session.js

```diff
@@ -155,6 +155,23 @@ function postToPortier(server: string, redirect: string): NavigationResult {
   }
 }
 
+export function navigateToOpenID(
+  authFormData: Object,
+  provider: Object
+): NavigationResult {
+  const { origin, pathname } = document.location;
+  const { server } = authFormData;
+  const strippedServer = server.replace(/\/$/, "");
+  const { auth_path } = provider;
+  const strippedAuthPath = auth_path.replace(/^\//, "");
+  const payload = btoa(JSON.stringify(authFormData));
+  const redirect = `${origin}${pathname}#/auth/${payload}/`;
+  document.location.href = `${strippedServer}/${strippedAuthPath}?callback=${encodeURIComponent(
+    redirect
+  )}&scope=openid email`;
```

Comment:
nit: maybe use intermediary variables for callback and scope to avoid wrapping?

Reply:
I think it's Pretty :)